File: backend/src/main/java/com/model/office/Item.java
package com.model.office;


import com.fasterxml.jackson.annotation.JsonIgnore;
import lombok.Data;

import javax.persistence.*;
import java.util.List;

@Entity
@Table(name = "items")
@Data
public class Item {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "item_id")
    private int id;

    @Column(name = "office_id")
    private int officeId;

    @Column(name = "item_count")
    private int count;

    @Column(name = "item_name")
    private String name;

    @JsonIgnore
    @OneToMany(mappedBy = "item", cascade = CascadeType.ALL, orphanRemoval = true)
    private List<RoomInventory> roomInventory;
}

File: frontend/src/app/model/item.ts
export class Item {
  id: number;
  officeId: number;
  count: number;
  name: string;

  constructor(item: Item) {
    this.id = item.id;
    this.officeId = item.officeId;
    this.count = item.count;
    this.name = item.name;
  }

  public toString = () => {
    return this.id + '/' + this.officeId + '/' + this.name + '(' + this.count + ')';
  }
}
